app/src/main/kotlin/at/bitfire/davdroid/sync/Syncer.kt

```diff
@@ -127,26 +127,32 @@ abstract class Syncer<CollectionType: LocalCollection<*>>(
 
     /**
      * Updates and deletes local collections. Specifically:
-     * - Deletes local collections if corresponding database collections are missing.
+     *
+     * - Deletes local collections if corresponding database collections are missing. If
+     * a local collection is removed because it's not in [dbCollections], **it becomes also removed from [localCollections]!**
```

Comment:
This is documenting the sideeffect, but it would be better not to have the sideeffect in the first place :)